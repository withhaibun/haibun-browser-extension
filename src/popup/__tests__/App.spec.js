import { shallowMount } from '@vue/test-utils'
import App from '../PopupApp'

const chrome = {
  storage: {
    local: {
      get: jest.fn(),
    },
  },
  extension: {
    connect: jest.fn(),
  },
}

describe('App.vue', () => {
  test('it has the correct pristine / empty state', () => {
    window.chrome = chrome
    const wrapper = shallowMount(App)
    expect(wrapper.element).toMatchSnapshot()
  })
})
