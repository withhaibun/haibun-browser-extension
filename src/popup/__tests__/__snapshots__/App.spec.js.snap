// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue it has the correct pristine / empty state 1`] = `
<div
  class="recorder"
  id="haibun-browser-extension"
>
  <div
    class="header"
  >
    <a
      href="#"
    >
       Headless recorder 
      <span
        class="text-muted"
      >
        <small>
          1.0.0
        </small>
      </span>
    </a>
    <div
      class="left"
    >
      <div
        class="recording-badge"
        style="display: none;"
      >
        <span
          class="red-dot"
        />
         recording
      </div>
      <button
        class="header-button"
      >
        <img
          alt="help"
          src=""
        />
      </button>
      <button
        class="header-button"
      >
        <img
          alt="settings"
          src=""
        />
      </button>
    </div>
  </div>
  <div
    class="main"
  >
    <div
      class="tabs"
    >
      <recording-tab-stub
        code=""
        is-recording="false"
        live-events=""
      />
      <div
        class="recording-footer"
      >
        <button
          class="btn btn-sm btn-primary"
        >
          Record
        </button>
        <button
          class="btn btn-sm btn-primary btn-outline-primary"
          style="display: none;"
        >
          Pause
        </button>
        <a
          href="#"
          style="display: none;"
        >
          view code
        </a>
        <checkly-badge-stub />
      </div>
      <!--v-if-->
      <div
        class="results-footer"
        style="display: none;"
      >
        <button
          class="btn btn-sm btn-primary"
          style="display: none;"
        >
           Restart 
        </button>
        <a
          href="#"
          style="display: none;"
        >
          copy to clipboard
        </a>
      </div>
    </div>
    <help-tab-stub
      style="display: none;"
    />
  </div>
</div>
`;
